lib/vault.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+const
+  crypto = require('crypto'),
+  fs = require('fs'),
+  _ = require('lodash');
+
+class Vault {
+  /**
+   * Prepare crypto primitives.
+   * Use the key passed in parameter
+   * environment variable.
+   *
+   * @param {string|null} vaultKey - key used to decrypt the secrets
+   */
+  constructor (vaultKey = null, secretsFile = null, encryptedSecretsFile = null) {
+    this.secrets = {};
+    this.secretsFile = secretsFile;
+    this.encryptedSecretsFile = encryptedSecretsFile;
+
+    if (_.isString(vaultKey) && vaultKey.length > 0) {
+      this.vaultKey = vaultKey;
```

Comment:
It looks like this property can be set after the object has been instantiated. In that case, shouldn't we add a setter for it, computing this.vaultKeyHash whenever a vaultKey is set?